lib/puppet/pops/types/class_loader.rb

```diff
@@ -103,7 +101,7 @@ def self.provide_from_name_path(name, name_path)
   end
 
   def self.find_class(name_path)
-    name_path.reduce(Object) do |ns, name|
+    name_path.inject(Object) do |ns, name|
```

Comment:
Always prefer `reduce` over `inject` (we use map/reduce as the paradigm instead of collect/inject).